spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/autoconfig/GcpCloudSqlAutoConfiguration.java

```diff
@@ -71,6 +75,9 @@
 	@Autowired
 	private GcpCloudSqlProperties gcpCloudSqlProperties;
 
+	@Autowired
+	private GcpProperties gcpProperties;
```

Comment:
They're different things. Here, `GcpProperties` refers to the core configuration, `GcpCloudSqlProperties` refers to the SQL configuration. It should be possible for both instances to contain different project IDs and credentials.

If you mean `GcpCloudSqlProperties` should contain a `GcpProperties` member instead of separate `projectId` and `credentialsLocation` members, then, as per http://docs.spring.io/spring-boot/docs/current-SNAPSHOT/reference/htmlsingle/#configuration-metadata-nested-properties, we'd end up with something like:

`spring.cloud.gcp.sql.gcp-properties:project-id` (similarly for `credentials-location`)

which I'm not sure I like..